packages/react-server/src/ReactFizzServer.js

```diff
@@ -97,119 +212,580 @@ function renderNode(request: Request, node: ReactNodeList): void {
+      // Therefore we'll need to assign it an ID - to refer to it by.
+      segment.id = request.nextSegmentId++;
+      // TODO: Move this to host config and format a placeholder node.
+      const instruction = new Uint8Array(0);
+      return writeChunk(destination, instruction);
+    }
+    case COMPLETED: {
+      segment.status = FLUSHED;
+      let r = true;
+      const chunks = segment.chunks;
+      let chunkIdx = 0;
+      const children = segment.children;
+      for (let childIdx = 0; childIdx < children.length; childIdx++) {
+        const nextChild = children[childIdx];
+        // Write all the chunks up until the next child.
+        for (; chunkIdx < nextChild.index; chunkIdx++) {
+          writeChunk(destination, chunks[chunkIdx]);
+        }
+        r = flushSegment(request, destination, nextChild);
+      }
+      // Finally just write all the remaining chunks
+      for (; chunkIdx < chunks.length; chunkIdx++) {
+        r = writeChunk(destination, chunks[chunkIdx]);
+      }
+      return r;
+    }
+    default: {
+      invariant(
+        false,
+        'Errored or already flushed boundaries should not be flushed again. This is a bug in React.',
+      );
+    }
+  }
+}
+
+function flushSegment(
+  request: Request,
+  destination,
+  segment: Segment,
+): boolean {
+  const boundary = segment.boundary;
+  if (boundary === null) {
+    // Not a suspense boundary.
+    return flushSubtree(request, destination, segment);
   }
+  boundary.parentFlushed = true;
+  // This segment is a Suspense boundary. We need to decide whether to
+  // emit the content or the fallback now.
+  if (boundary.forceClientRender) {
+    // Emit a client rendered suspense boundary wrapper.
+    // We never queue the inner boundary so we'll never emit its content or partial segments.
+
+    // TODO: Move this to host config.
+    const boundaryStart = new Uint8Array(0);
+    writeChunk(destination, boundaryStart);
+
+    // Flush the fallback.
+    flushSubtree(request, destination, segment);
+
+    const boundaryEnd = new Uint8Array(0);
+    return writeChunk(destination, boundaryEnd);
+  } else if (boundary.pendingWork > 0) {
+    // This boundary is still loading. Emit a pending suspense boundary wrapper.
 
-  ReactCurrentDispatcher.current = prevDispatcher;
+    // Assign an ID to refer to the future content by.
+    boundary.rootSegmentID = request.nextSegmentId++;
+    if (boundary.completedSegments.length > 0) {
+      // If this is at least partially complete, we can queue it to be partially emmitted early.
+      request.partialBoundaries.push(boundary);
+    }
+
+    // TODO: Move this to host config.
+    const boundaryStart = new Uint8Array(0);
+    writeChunk(destination, boundaryStart);
+
+    // Flush the fallback.
+    flushSubtree(request, destination, segment);
+
+    const boundaryEnd = new Uint8Array(0);
+    return writeChunk(destination, boundaryEnd);
+  } else if (boundary.byteSize > request.maxBoundarySize) {
```

Comment:
This is a really neat feature

Reply:
This is how you'd take advantage of streaming even with SSG.